metadata/params/params.go

```diff
@@ -71,7 +105,22 @@ func visitAllParams(components ast.Object) (map[string]Params, error) {
 		if err != nil {
 			return nil, err
 		}
-		params[string(*f.Id)] = p
+		var id string
+		switch f.Kind {
+		case ast.ObjectFieldStr:
```

Comment:
Am I missing something, or does it make sense to just have `hasComponent` return a string as well? This seems very similar?